test/test_data/configure_check.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <string_view>
+#include <unordered_map>
+
+#include "proxy_wasm_intrinsics.h"
+
+class TestRootContext1 : public RootContext {
```

Comment:
Is there value in having separate pairs of Test*Context1 vs. Test*Context2 classes? Alternatively this could just define a single pair of classes, TestRootContext and TestContext, and have onConfigure log the constructor-provided root_id value. There would still be two RegisterContextFactory instantiations with "root_id_1" and "root_id_2", but they would both use CONTEXT_FACTORY(TestContext) and ROOT_FACTORY(TestRootContext).

Or am I misunderstanding and this would sacrifice some aspect of the test?

Reply:
You are right. 

I thought that using the same context with the different root context is unusual, so, for the demonstration purpose,  I used the separate contexts: `TestContext1` and `TestContext2`. 

BTW, Done as you commented.